bpo-1104: msilib.SummaryInfo.GetProperty() truncates the string by one character

uranusjr opened the pull request:
Based on the patch in bpo-1104 by Anthony Tuininga (atuining) and Mark Mc Mahon (markm).

> Unfortunately Microsoft has deemed it necessary to return the size
> of the string without the null termination character but insists
> upon the size including it when passing it in. Arggh!
>
> — Anthony, on the fix

<!-- issue-number: bpo-1104 -->
https://bugs.python.org/issue1104
<!-- /issue-number -->

uranusjr committed:
Add one char to MsiSummaryInfoGetProperty() output

Based on the patch in bpo-1104 by Anthony Tuininga (atuining) and Mark
Mc Mahon (markm).

    Unfortunately Microsoft has deemed it necessary to return the size
    of the string without the null termination character but insists
    upon the size including it when passing it in. Arggh!

                                            -- Anthony, on the fix

the-knights-who-say-ni commented:
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

uranusjr committed:
Improve malloc/free call in summary_getproperty()

This makes the malloc/free calls more trackable even when more types are
added to MsiSummaryInfoGetProperty.

zooba committed:
Merge branch 'master' into bpo-1104-msilib-getproperty-truncation
